src/common/context/OnboardContext.tsx

```diff
@@ -191,12 +201,94 @@ export const OnboardContext = createContext<TOnboardContext>(
 
 OnboardContext.displayName = "OnboardContext";
 
+// ensure we dont instanciate notify more than once per chain, cache the instance
+const NotifyFactory = () => {
+  const instances = new Map<number, NotifyAPI>();
+  return (networkId: number = 1) => {
+    if (instances.has(networkId)) return instances.get(networkId) as NotifyAPI;
+    const instance = Notify({
+      dappId: process.env.REACT_APP_PUBLIC_ONBOARD_API_KEY, // [String] The API key created by step one above
+      networkId, // [Integer] The Ethereum network ID your Dapp uses.
+      desktopPosition: "topRight",
+    });
+    instances.set(networkId, instance);
+    return instance;
+  };
+};
+
+const getNotify = NotifyFactory();
+
 const OnboardProvider: FC<WithDelegatedProps> = ({
   children,
   ...delegated
 }) => {
   const [state, dispatch] = useReducer(connectionReducer, INITIAL_STATE);
 
+  useEffect(() => {
+    events.on("event", (event: string, data: any) => {
+      switch (event) {
```

Comment:
this code is almost identically copied from the previous subscription. some uneeded network calls are removed from the wallet event